tests/base/converters/test_dataframe_to_mds.py

```diff
@@ -111,23 +119,30 @@ def test_end_to_end_conversion_local_nocolumns(self, dataframe: Any, keep_local:
                               merge_index=merge_index,
                               mds_kwargs=mds_kwargs)
 
-        assert (len(os.listdir(out)) > 0), f'{out} is empty'
-        for d in os.listdir(out):
-            if os.path.isdir(os.path.join(out, d)):
-                assert (os.path.exists(os.path.join(
-                    out, d, 'index.json'))), f'No index.json found in subdirectory {d}'
+        if keep_local:
+            assert (len(os.listdir(out)) > 0), f'{out} is empty'
+            for d in os.listdir(out):
+                if os.path.isdir(os.path.join(out, d)):
+                    assert (os.path.exists(os.path.join(
+                        out, d, 'index.json'))), f'No index.json found in subdirectory {d}'
 
         if merge_index:
-            assert (os.path.exists(os.path.join(out, 'index.json'))), 'No merged index.json found'
-            mgi = json.load(open(os.path.join(out, 'index.json'), 'r'))
-            nsamples = 0
-            for d in os.listdir(out):
-                sub_dir = os.path.join(out, d)
-                if os.path.isdir(sub_dir):
-                    shards = json.load(open(os.path.join(sub_dir, 'index.json'), 'r'))['shards']
-                    if shards:
-                        nsamples += shards[0]['samples']
-            assert (nsamples == sum([a['samples'] for a in mgi['shards']]))
+            if keep_local:
+                assert (os.path.exists(os.path.join(out,
```

Comment:
asserts don't need parens (here and elsewhere)

Reply:
removed